lexibank_chenhmongmien.py

```diff
@@ -34,90 +33,63 @@ class Dataset(NonSplittingDataset):
 
-        vob_table = [x for x in vob_table if x != []]
-
-        with open(self.dir.joinpath("raw", "languages.csv").as_posix(), "w", newline="") as lw:
-            languagewriter = csv.writer(lw, delimiter=",", quotechar='"')
-            languagewriter.writerow(language_table_header)
-            languagewriter.writerows(language_table)
-            lw.close()
-
-        with open(self.dir.joinpath("raw", "raw.csv").as_posix(), "w", newline="") as vw:
-            vocabwriter = csv.writer(vw, delimiter=",", quotechar='"')
-            vocabwriter.writerow(vob_table_header)
-            vocabwriter.writerows(vob_table)
-            vw.close()
+        self.raw_dir.write_csv(
+            'raw.csv',
+            [r for r in iter_rows(soup.findAll("table", {"class": "wikitable sortable"})[1]) if r])
 
     def clean_form(self, item, form):
         if form not in ["*", "---", "-"]:
             form = strip_brackets(split_text(form, separators=";,/")[0])
             return form.replace(" ", "_")
 
-    def cmd_install(self, **kw):
+    def cmd_makecldf(self, args):
         """
         Convert the raw data to a CLDF dataset.
         """
-
-        with open(self.dir.joinpath("raw", "raw.csv").as_posix(), "r") as csvfile:
-            reader = csv.DictReader(csvfile, delimiter=",", quotechar='"')
-            data = [row for row in reader]
+        data = self.raw_dir.read_csv('raw.csv', dicts=True)
         languages, concepts = {}, {}
-        with self.cldf as ds:
-            for concept in self.conceptlist.concepts.values():
-                ds.add_concept(
-                        ID=concept.number,
-                        Name=concept.gloss,
-                        Concepticon_ID=concept.concepticon_id,
-                        Concepticon_Gloss=concept.concepticon_gloss,
-                        Chinese_Gloss=concept.attributes['chinese']
-                )
-                concepts[concept.attributes['chinese']] = concept.number
-
-            ds.add_languages()
-            languages = {k['Name']: k['ID'] for k in self.languages}
 
-            ds.add_sources(*self.raw.read_bib())
-            missing = {}
-            for cgloss, entry in tqdm(enumerate(data), desc='cldfify the data', total=len(data)):
-                if entry['Chinese gloss'] in concepts.keys():
-                    for language in languages:
-                        value = self.lexemes.get(entry[language], entry[language])
-                        if value.strip():
-                            ds.add_lexemes(
-                                Language_ID=languages[language],
-                                Parameter_ID=concepts[entry['Chinese gloss']],
-                                Value=value,
-                                Source=['Chen2013'],
-                            )
-                else:
-                    missing[entry['Chinese gloss']] += 1
+        for concept in self.conceptlist.concepts.values():
+            args.writer.add_concept(
```

Comment:
What used to be obtained as `with self.cldf` is now available as `args.writer` (and the context is already entered).

Reply:
Ah, very nice. That's making the code much more readable.